main.py

```
@@ -53,219 +76,919 @@
+            status="submitted",
+            report_id=report.report_id,
+            timestamp=timestamp
         )
         
+    except HTTPException:
+        raise
     except Exception as e:
-        logger.error(f"Image analysis failed: {str(e)}")
-        raise HTTPException(status_code=500, detail=f"Image analysis failed: {str(e)}")
+        logger.error(f"Report submission error: {e}")
+        raise HTTPException(status_code=500, detail=f"Report submission failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)